kong/dao/migrations/helpers.lua

```diff
@@ -0,0 +1,94 @@
+local _M = {}
+local json_decode = require("cjson.safe").decode
+
+
+-- Iterator to update plugin configurations.
+-- It works indepedent of the underlying datastore.
+-- @param dao the dao to use
+-- @param plugin_name the name of the plugin whos configurations
+-- to iterate over
+-- @return `ok+config+update` where `ok` is a boolean, `config` is the plugin configuration
+-- table (or the error if not ok), and `update` is an update function to call with
+-- the updated configuration table
+-- @usage
+--    up = function(_, _, dao)
+--      for ok, config, update in plugin_config_iterator(dao, "jwt") do
+--        if not ok then
+--          return config
+--        end
+--        if config.run_on_preflight == nil then
+--          config.run_on_preflight = true
+--          local _, err = update(config)
+--          if err then
+--            return err
+--          end
+--        end
+--      end
+--    end
+function _M.plugin_config_iterator(dao, plugin_name)
+
+  -- iterates over rows
+  local run_rows = function(t)
+    for _, row in ipairs(t) do
+      if type(row.config) == "string" then
+        -- de-serialize in case of Cassandra
+        local json, err = json_decode(row.config)
+        if not json then
+          return nil, ("json decoding error '%s' while decoding '%s'"):format( 
+                      tostring(err), tostring(row.config))
+        end
+        row.config = json
+      end
+      coroutine.yield(row.config, function(updated_config)
+        if type(updated_config) ~= "table" then
+          return nil, "expected table, got " .. type(updated_config)
+        end
+        row.created_at = nil
+        row.config = updated_config
+        return dao.plugins:update(row, {id = row.id})
+      end)
+    end
+  end
+
+  local coro
+  if dao.db_type == "cassandra" then
+    coro = coroutine.create(function()
+      for rows, err in dao.db.cluster:iterate([[
```

Comment:
The recent changes in how we handle migrations in Cassandra introduced a new rule of thumb: we only run migrations on a single coordinator (whether they are DDL or DML queries), and let the elected coordinator propagate the changes to its peers via the inter-nodes Cassandra gossiping.

The `cluster:iterate()` method will load-balance each query and thus, request each page from a different coordinator. The fix is very simple.

Instead, we should use the coordinator that has been elected to run the migrations. This has been done via `dao.db:first_coordinator()` in [factory.lua#L378](https://github.com/Mashape/kong/blob/master/kong/dao/factory.lua#L378). We'll need a new method to retrieve this coordinator, since it is stored in an [upvalue](https://github.com/Mashape/kong/blob/master/kong/dao/db/cassandra.lua#L180). Then, we can simply do:
```lua
for rows, err in coordinator:iterate() do

end
```

Because the `iterate()` API is the same whether it is used in the `cassandra` or `cassandra.cluster` modules.

Reply:
good catch!